autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java

```diff
@@ -182,8 +195,19 @@ private void publish(TimeAlert alert) {
         event.setExistingClusterNodeCount(stackV4Response.getNodeCount());
         event.setScalingAdjustmentType(REGULAR);
         scalingActivityMsg = messagesService.getMessageWithArgs(AUTOSCALE_SCHEDULE_BASED_UPSCALE, targetIncrementNodeCount);
+
+        Set<String> servicesRunningOnHostgroup = stackResponseUtils.getServicesOnHostGroup(stackV4Response, alert.getScalingPolicy().getHostGroup());
+        if (servicesRunningOnHostgroup.contains(IMPALA) && !servicesRunningOnHostgroup.contains(YARN)) {
+            if (alert.getScalingPolicy().getHostGroup().equalsIgnoreCase(HOSTGROUP_COORDINATOR)) {
+                throw new BadRequestException("Schedule based Scaling in IMPALA is not allowed on coordinator node.");
+            }
+            alertValidator.validateImpalaScheduleBasedScalingEntitlement(alert.getCluster().getStackCrn(),
```

Comment:
The AlertValidator should not be invoked inside of the CronTimeEvaluator. Blueprints are fairly static. When autoscaling is enabled for a cluster - we should be able to extract the services on a node at that point, and store them.
So
- Controller calls to enable/disable autoscale - perform the check for the services running, and the impala entitlement. Note: This call cannot go to CM - it has to be handled via the CB blueprint. (Can we extract co-ordinator vs executor from the CB blueprint?)
- They, additionally, extract out the services running on the host group and store them.
- The CronTimeEvaluator code looks up these stored services to make decisions on whether YARN, IMPALA or Both (Cleaning this up / abstracting it comes in a follow up patch)